flox-bash/lib/bootstrap.sh

```diff
@@ -25,7 +25,7 @@ function bootstrap() {
 	floxUserMetaRegistry get floxClientUUID >/dev/null || \
 		floxUserMetaRegistry set floxClientUUID $($_uuid)
 	floxClientUUID=$(floxUserMetaRegistry get floxClientUUID)
-	if [ -t 1 ]; then
```

Comment:
I think that breaks the "non-interactive" `flox activate`, as in:

`$ eval $(flox activate)` 

run in an interactive shell session (that includes `.bashrc/.zshrc`).

IMO there is a difference between an "interactive" that changes output and formatting (i.e. how git/ls/cargo/nix output changes when run in ci) and an "interactive" whare a human _could_ answer to prompts.

Those are not necessarily the same: In `flox activate` we do want to detect if the output is collected but we still want to be able to prompt users.

Reply:
would you mind checking if [55fe699](https://github.com/flox/flox/pull/112/commits/55fe699d1726f862a72792489040476b6b0d0bba) changes the right references to $interactive?

There is one left that seems to fall in the "logging" kind of interaction so i left that the same

Reply:
I think so .. and I was working on the same patch on this end! My version was a little more explicit though by creating a new `$spawnMode` variable - I think the logic is more clear and will prevent us from conflating these concepts again by giving it a name. Can you have a look at 741ac9be0d0755760f494df4cc6f1fc2e79b1a79 to see if that looks more clear?